core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -120,14 +120,25 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
             val ret = mesosDriver.run()
             logInfo("driver.run() returned with code " + ret)
             if (ret != null && ret.equals(Status.DRIVER_ABORTED)) {
-              error = Some(new SparkException("Error starting driver, DRIVER_ABORTED"))
-              markErr()
+              val ex = new SparkException("Error starting driver, DRIVER_ABORTED")
+              // if the driver gets aborted after the successful registration
```

Comment:
Also to simplify the code, can we just throw SparkExecption here? Then the catch will then handle all cases